Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -27,6 +31,25 @@ using model::Document;
 using model::DocumentKey;
 using model::ResourcePath;
 
+namespace {
+
+// Convert a vector of unique_ptr's to a vector of shared_ptr's
+std::vector<std::shared_ptr<core::Filter>> Convert(
+    std::vector<std::unique_ptr<core::Filter>>&& v) {
+  std::vector<std::shared_ptr<core::Filter>> result;
+  std::move(v.begin(), v.end(), std::back_inserter(result));
+  v.clear();
+  return result;
+}
+
+}  // namespace
+
+Query::Query(ResourcePath path,
+             std::vector<std::unique_ptr<core::Filter>>&&
```

Comment:
Given that we only ever want Query to contain a vector of `std::shared_ptr`, why not just accept that type here?

If there's some wild client of query that needs to make this conversion make it there problem. There's no reason for us to make this easy.

Reply:
Done.